frontend/components/Settings/WhartonApplicationCycles.tsx

```diff
@@ -0,0 +1,300 @@
+    if (clubsToRemove.length > 0) {
+      doApiRequest(`/cycles/remove_clubs_from_all/`, {
+        method: 'POST',
+        body: { clubs: clubsToRemove.map((x) => x.value) },
+      })
+    }
+    if (clubsToAdd.length > 0) {
+      doApiRequest(`/cycles/${membershipCycle.id}/add_clubs/`, {
+        method: 'POST',
+        body: { clubs: clubsToAdd.map((x) => x.value) },
+      })
+    }
+  }
+
+  const closeExtensionsModal = (): void => {
+    setEditExtensions(false)
+    // calculate clubs that have changed
+    const clubsToUpdate = clubsExtensions.filter((x) => x.changed)
+    // split into clubs with exceptions and clubs without
+    const clubsExceptions = clubsToUpdate.filter((x) => x.exception)
+    const clubsNoExceptions = clubsToUpdate.filter((x) => !x.exception)
+
+    // call /cycles/:id/add_clubs and /cycles/remove_clubs_from_all with data.clubs as list of ids
+    if (clubsExceptions.length > 0) {
+      doApiRequest(`/cycles/add_clubs_to_exception/`, {
+        method: 'POST',
+        body: {
+          clubs: clubsExceptions.map((x) => {
+            // eslint-disable-next-line camelcase
+            return { id: x.id, end_date: x.endDate }
+          }),
+        },
+      })
+    }
+    if (clubsNoExceptions.length > 0) {
+      doApiRequest(`/cycles/remove_clubs_from_exception/`, {
+        method: 'POST',
+        body: { clubs: clubsNoExceptions.map((x) => x.id) },
+      })
+    }
+  }
+
+  useEffect(() => {
+    doApiRequest('/whartonapplications/?format=json')
+      .then((resp) => resp.json())
+      .then((data) => {
+        setClubOptionsMembership(
+          data.map((club: ClubApplication) => {
+            return { label: club.name, value: club.id }
+          }),
+        )
+      })
+  }, [])
+
+  useEffect(() => {
+    if (membershipCycle && membershipCycle.id != null) {
+      doApiRequest(`/cycles/${membershipCycle.id}/clubs?format=json`)
+        .then((resp) => resp.json())
+        .then((data) => {
+          const initialOptions = data.map((club: ClubApplication) => {
+            return { label: club.name, value: club.id }
+          })
+          setClubsInitialOptionsMembership(initialOptions)
+          setClubsSelectedMembership(initialOptions)
+        })
+    }
+  }, [membershipCycle])
+
+  useEffect(() => {
+    if (extensionsCycle && extensionsCycle.id != null) {
+      doApiRequest(`/cycles/${extensionsCycle.id}/clubs?format=json`)
+        .then((resp) => resp.json())
+        .then((data) => {
+          const initialOptions = data.map((club: ClubApplication) => {
+            return {
+              id: club.id,
+              clubName: club.name,
+              endDate: new Date(club.application_end_time),
+              exception: club.application_end_time_exception,
+              changed: false,
+            }
+          })
+          setClubsExtensions(initialOptions)
+        })
+    }
+  }, [extensionsCycle])
+
+  if (clubOptionsMembership == null) {
```

Comment:
Suggestion: We should try to use something like react-query or SWR to streamline loading process